### PR TITLE
Extend sublime-snippets schema to support kind info.

### PR DESCRIPTION
**Pull request by deathaxe**

### CODE REVIEW

File: schemas/sublime-completions.json [hunk 1]
```diff
@@ -9,19 +9,43 @@
...
   "properties": {
     "scope": {
       "type": "string",
-      "description": "A selector to match the scope name for completions to be enabled"
+      "description": "A selector to match the scope name for completions to be enabled."
     },
     "completions": {
       "type": "array",
-      "description": "A list of completion objects with trigger and contents. Alternatively a list of plain completion strings.",
+      "markdownDescription": "**A list of completions**\n\nConsiting of plain completion strings:\n\n```js\n[\n    \"function\",\n    \"variable\\tannotation\"\n]\n```\n\nor objects with trigger and contents:\n\n```js\n[\n    { \"trigger\": \"function(arg1, arg2)\", \"contents\": \"function($1, ${2:arg2})\" },\n    { \"trigger\": \"variable\\tannotation\", \"contents\": \"variable=\" }\n]\n```",
```
Comment (rchl) [inline schemas/sublime-completions.json:43]: Typo: "Consiting"

File: schemas/sublime-completions.json [hunk 2]
```diff
@@ -9,19 +9,43 @@
...
       "items": {
         "oneOf": [
           {
             "type": "string",
-            "description": "A completion string that is at the same time a trigger for that completion"
+            "markdownDescription": "**Completion String**\n\nA plain string which is both a trigger and the contents to add to the editor.\nIt can optionally be followed by a tab character (`\\t`) and an annotation.\n\n```js\n[\n    \"function\",\n    \"variable\\tannotation\"\n]\n```\n\n**Annotations:**\n\n- are displayed on the right-hand edge of the completions popup,\n  and may contain any information the author deems useful.\n- typically consist of a word or two.\n- are not added to the editor when committing the comletion."
```
Comment (rchl) [inline schemas/sublime-completions.json:48]: Typo: "comletion"

File: schemas/sublime-completions.json [hunk 3]
```diff
@@ -32,11 +56,44 @@
...
+                "oneOf": [
+                    {
+                      "type": "array",
+                      "items": [
+                        {
+                          "markdownDescription": "**Kind Category**\n\nA highlevel category of a symbol kind,\nwhich is used by the theme to select the color of the kind metadata.\n\nIt must be one of: \n\n- `\"ambiguous\"`\n- `\"function\"`\n- `\"keyword\"`\n- `\"markup\"`\n- `\"namespace\"`\n- `\"navigation\"`\n- `\"snippet\"`\n- `\"type\"`\n- `\"variable\"`",
```
Comment (rchl) [inline schemas/sublime-completions.json:80]: - Maybe add a space in "highlevel"
- Can we skip the double quotes around category types to make it look cleaner?

File: schemas/sublime-completions.json [hunk 4]
```diff
@@ -32,11 +56,44 @@
...
+                          "$ref": "#/definitions/kindCategory"
+                        },
+                        {
+                          "markdownDescription": "**Kind Icon**\n\nA single Unicode character to be shown to the left of the trigger",
```
Comment (rchl) [inline schemas/sublime-completions.json:84]: Missing period at the end.

File: schemas/sublime-completions.json [hunk 5]
```diff
@@ -32,11 +56,44 @@
             "properties": {
               "trigger": {
                 "type": "string",
-                "markdownDescription": "Text that will be displayed in the completions list and will cause the contents to be inserted when chosen.\nYou can optionally follow it with a tab character (`\\t`) and a text to add a hint for the preceding trigger. The hint will be displayed alongside the trigger and does not affect the trigger itself."
+                "markdownDescription": "**Completion Trigger**\n\nThe text the user must enter to match the completion.\nIt can optionally be followed by a tab character (`\\t`) and an annotation.\n\n```js\n[\n    { \"trigger\": \"function\", ... },\n    { \"trigger\": \"variable\\tannotation\", ... },\n]\n```\n\n**Annotations:**\n\n- are displayed on the right-hand edge of the completions popup,\n  and may contain any information the author deems useful.\n- typically consist of a word or two.\n- can also be defined by dedicated `annotations` field _(requires ST4073)_."
```
Comment (rchl) [inline schemas/sublime-completions.json:59]: This string mentions that ST4073 is required for one thing but all these features are ST4+ really so IMO we don't need that note.

We should either just completely omit that detail or annotate all ST4-only features properly.

File: schemas/sublime-completions.json (hunk 3 above)
Comment (deathaxe) [reply schemas/sublime-completions.json]: I'll use `high-level` as it is the term in the official completions docs.

Comment (rchl) [reply schemas/sublime-completions.json]: Yes, sorry, that's even better.

**Pull request merged**